module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -82,7 +86,35 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 		})
 		a.keeper.AfterSignerSetExecutedEvent(ctx, *event)
 		return nil
+	case *types.SendToIBCEvent:
+		// Check if coin is Cosmos-originated asset and get denom
+		isCosmosOriginated, denom := a.keeper.ERC20ToDenomLookup(ctx, event.TokenContract)
+		addr, _ := sdk.AccAddressFromBech32(event.CosmosReceiver)
+		coin := sdk.NewCoin(denom, event.Amount)
+		coins := sdk.Coins{coin}
+
+		if !isCosmosOriginated {
+			if err := a.DetectMaliciousSupply(ctx, denom, event.Amount); err != nil {
+				return err
+			}
+
+			// if it is not cosmos originated, mint the coins (aka vouchers)
+			if err := a.bankKeeper.MintCoins(ctx, types.ModuleName, coins); err != nil {
+				return sdkerrors.Wrapf(err, "mint vouchers coins: %s", coins)
+			}
+		}
 
+		if err := a.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr, coins); err != nil {
+			return err
+		}
+
+		// format of this could be {port}/{channel}
+		chanport := strings.Split(event.Channel, "/")
+		if err := a.transferKeeper.SendTransfer(ctx, chanport[0], chanport[1], coin, addr, event.ForwardAddress, clienttypes.NewHeight(0, 0), uint64(time.Second*100)); err != nil {
```

Comment:
One quick note, would be cool to add an extra parameter here to allow for gov configurable fee to be applied and funds to be put into fee pool. I think that would be pretty quick to implement too. cc @zmanian